compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -757,13 +758,18 @@ trait Applications extends Compatibility {
               }
               else defaultArgument(normalizedFun, n, testOnly)
 
+            // a bug allowed empty parens to expand to implicit args: fail empty args for rewrite on migration
+            def canSupplyImplicits = methodType.isImplicitMethod
+              && (applyKind == ApplyKind.Using || false.tap: _ =>
```

Comment:
This `false.tap` looks pretty cool, but IMHO it is pretty hard to read. Can you change it into a block with a `false` value?

Reply:
Agree it's too much mechanism in a boolean expr.